templates/launch.html

```diff
@@ -158,19 +158,16 @@ <h1>Launch a survey</h1>
 
     var refresh = document.getElementsByClassName('refresh');
 
-    function rando(el) {
-        var result = '';
-        var length = '8';
-        var chars = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
-        var el = el;
-        for (var i = length; i > 0; --i) result += chars[Math.round(Math.random() * (chars.length - 1))];
-        document.getElementById(el).value = result;
+    function uuidv4() {
+      return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
+        (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
+      )
     }
 
     for (var i = 0; i < refresh.length; i++) {
         refresh[i].addEventListener("click", function() {
             var el = this.getAttribute("data-refresh");
-            rando(el);
+            document.getElementById(el).value = uuidv4();
```

Comment:
Could this be called when the page gets rendered too? At the moment it's just showing the default "789" until the refresh button is clicked.